### DISCUSSION TITLE
Loses connectivity (seemingly) at random.

### DISCUSSION THREAD

**Discussion by shiggityputz** [darthcloud/BlueRetro#204] [Q&A] [answered]
This is related to this discussion, just fyi: https://github.com/darthcloud/BlueRetro/discussions/177#discussioncomment-2217015

I was able to get the device working properly, but decided to add a power switch to it much like in the video that Yakara had shown and installing the board internally.  To be clear, it worked completely fine.  But today I wanted to do the power switch install, and also place some protective kapton/electrical tape inside the console just in case.

The power switch is definitely installed correctly (i can cut power to it, or turn it back on, etc.)  Now, the ESP32 board WILL connect to a controller and it works flawlessly, but it seems to lose connection when moving it or placing it where it needs to go in the console. I figured maybe my solder joints weren't great, so I went and redid them to be safe. They arent the most beautiful solder joints in the world, but they are solid, and do not make any bridges.

I pulled a log from it like instructed in the other thread, and I came up with this:  https://controlc.com/937920d3

Im not sure if the placement is what cold be causing the issue, but up until today, the board was sitting in the same position. The only thing that changed is that I removed the electrical tape, replaced it with kapton tape, and added the power switch.

Some images for more info: https://imgur.com/a/J1Mk28z

**Comment by HornheaDD** [comment]
Sorry - I think I signed in with an old account, that should have been posted by HornheaDD.  Apologies.

**Comment by darthcloud** [comment, marked as answer]
I count 8 wire, so you hook up only 3 ports?

If not wiring 4th port plug, make sure to connect IO26 (P4_D0) & IO27 (P4_D1) to 3.3V

Not doing this make the pin unstable and so it might work sometimes and sometimes not.

Your log show the esp32 reboot in loop due to watchdog, which is always caused by a unused port pins left floating.

**Reply by HornheaDD** [reply]
Hm, ok. Well, I *did* have the 4th controller port wired initially, but this was still happening. When I checked the console to see how many controllers are being identified, (by going into the memory card menu, it will list any controllers currently connected) it only listed 3. The 4th port was connected, but it was still happening.  So I went ahead and just removed the 4th controller. I will wire the 4th controller to 3.3v and report back.  I dont know if this helps, but I recorded a video of what is happening. It looks like it might be unstable like you mentioned.  Here's the vid: https://youtu.be/XFl8X5_QvoM

**Reply by HornheaDD** [reply]
Welp, that seems to have been the fix. Weird! Thanks again dude!
